### ISSUE TITLE
Add an easy way to compute loss during validation

### ISSUE THREAD

**Issue by KevinMusgrave** [KevinMusgrave/pytorch-metric-learning#433]
It can be useful to compute the average loss over the training set or validation set, and use that (rather than some accuracy metric) to select the best checkpoint. See https://github.com/KevinMusgrave/pytorch-metric-learning/issues/122#issuecomment-1050649626 for a discussion.
